File: android/src/main/java/com/margelo/nitro/visioncamerafacedetection/HybridFaceScannerFactory.kt
package com.margelo.nitro.visioncamerafacedetection

import androidx.annotation.Keep
import com.facebook.proguard.annotations.DoNotStrip
import com.margelo.nitro.camera.HybridCameraOutputSpec

@DoNotStrip
@Keep
class HybridFaceScannerFactory : HybridFaceScannerFactorySpec() {
  @DoNotStrip
  @Keep
  override fun createFaceScanner(options: FaceScannerOptions): HybridFaceScannerSpec {
    return HybridFaceScanner(options)
  }

  @DoNotStrip
  @Keep
  override fun createFaceScannerOutput(options: FaceScannerOutputOptions): HybridCameraOutputSpec {
    return HybridFaceScannerOutput(options)
  }
}

File: ios/HybridFaceScannerFactory.swift
import NitroModules
import VisionCamera

class HybridFaceScannerFactory: HybridFaceScannerFactorySpec {
  func createFaceScanner(options: FaceScannerOptions) throws -> any HybridFaceScannerSpec {
    return HybridFaceScanner(options: options)
  }
  
  func createFaceScannerOutput(options: FaceScannerOutputOptions) throws
  -> any HybridCameraOutputSpec
  {
    return HybridFaceScannerOutput(options: options)
  }
}
